aten/src/ATen/native/cuda/BatchLinearAlgebra.cu

```diff
@@ -2108,65 +2117,68 @@ AT_ERROR("symeig: MAGMA library not found in "
   magma_int_t iwkopt;
   magma_int_t lrwork = -1;
   value_t rwkopt;
-  magmaSymeig<scalar_t, value_t>(jobz, uplo, n, self_data, n, eigvals_data, wA, n, &wkopt, lwork, &rwkopt, lrwork, &iwkopt, liwork, &info);
+  magmaSyevd<scalar_t, value_t>(jobz, uplo, n, vectors_data, lda, values_data,
+    wA, lda, &wkopt, lwork, &rwkopt, lrwork, &iwkopt, liwork, infos_data);
 
   scalar_t* work;
   magma_int_t* iwork;
-  lwork = magma_int_cast(real_impl<scalar_t, value_t>(wkopt), "work_size");
-  liwork = magma_int_cast(iwkopt, "iwork_size");
+  lwork = std::max<magma_int_t>(1, magma_int_cast(real_impl<scalar_t, value_t>(wkopt), "work_size"));
+  liwork = std::max<magma_int_t>(1, magma_int_cast(iwkopt, "iwork_size"));
   ALLOCATE_ARRAY(work, scalar_t, lwork);
   ALLOCATE_ARRAY(iwork, magma_int_t, liwork);
 
   value_t* rwork = nullptr;
   c10::Storage storage_rwork;
-  if (isComplexType(at::typeMetaToScalarType(self.dtype()))) {
-    lrwork = magma_int_cast(rwkopt, "rwork_size");
+  if (vectors.is_complex()) {
+    lrwork = std::max<magma_int_t>(1, magma_int_cast(rwkopt, "rwork_size"));
     storage_rwork = pin_memory<value_t>(lrwork);
     rwork = static_cast<value_t*>(storage_rwork.data());
   }
 
-  for (int64_t i = 0; i < batch_size; i++) {
-    scalar_t* self_working_ptr = &self_data[i * self_matrix_stride];
-    value_t* eigvals_working_ptr = &eigvals_data[i * eigvals_stride];
-    magmaSymeig<scalar_t, value_t>(jobz, uplo, n, self_working_ptr, n, eigvals_working_ptr,
-                          wA, n, work, lwork, rwork, lrwork, iwork, liwork, &info);
-    infos[i] = info;
-    if (info != 0) {
+  for (decltype(batch_size) i = 0; i < batch_size; i++) {
```

Comment:
prefer c10::irange

Reply:
I also prefer that, but I had to remove it in aa965d9bb2e1705825010857fc81f0b958e131a3 because apparently nvcc from CUDA 10 doesn't like `c10::irange`. See compilation failure here https://app.circleci.com/pipelines/github/pytorch/pytorch/279513/workflows/bef49946-0ae0-4e99-a24f-62267d8b2b9b/jobs/11228805
It works fine with CUDA 11.

Reply:
hmm, I was getting the same errors. I fixed them by explicitly defining the template types which can be a pain haha.